packages/okta-react/src/ImplicitCallback.js

```diff
@@ -26,18 +25,23 @@ export default withAuth(class ImplicitCallback extends Component {
 
   componentDidMount() {
     this.props.auth.handleAuthentication()
-    .then(() => this.setState({ authenticated: true }))
+    .then(() => {
+      const location = this.props.auth.getFromUri();
+      this.setState({ authenticated: true, location });
+    })
     .catch(err => this.setState({ authenticated: false, error: err.toString() }));
   }
 
   render() {
-    if (this.state.authenticated === null) {
-      return null;
+    const { error, authenticated, location } = this.state;
+    if (error) {
```

Comment:
is `error` ever temporarily populated?

Reply:
The logic here should be unchanged from previous behavior. `handleAuthentication()` is called once on `componentDidMount`, if that call throws, the error is put into state where it can be rendered. There is no logic to remove it from state, so it will persist if the component is repeatedly rendered.

Reply:
The order of the logic is different.

Previous:
- If not auth, then null.  Otherwise, if auth succeeded redirect, else show error

Now:
- If error, show error.  Otherwise, if auth and location, redirect.  

An early population of error previously had no effect.  Now it's the first thing checked.

Reply:
What do you mean by "early population"? `error` and `authenticated` are set at the same time, after the result from `handleAuthentication` I don't see a case where `authenticated` would be null but error would be set. If error is set, I expect `authenticated` to be `false`. Is there another way that `error` can be set, besides the catch block on line 32?

Reply:
That was my question.  If that's not a problem, I'm good.